src/SpectralDataManager.cpp

```diff
@@ -0,0 +1,214 @@
+#include <iostream>
+#include "FFT.h"
+#include "SpectralDataManager.h"
```

Comment:
I recommend this rule:  in a .cpp file, the first `#include` directive should always be for the corresponding .h file.  Putting it before any others causes that header file to be properly self-contained -- you can be sure it is not depending on some other header being included first to make it compilable.  Instead you are forced to make all necessary nested include's or make forward declarations of incomplete types.

Maybe you know this rule -- you followed it in all other examples of new .cpp files, but just forgot it once here.